src/nvim/os/env.c

```diff
@@ -119,6 +119,9 @@ int os_setenv(const char *name, const char *value, int overwrite)
   if (!overwrite && os_getenv(name) != NULL) {
     return 0;
   }
+  if (value[0] == '\0') {
+    return os_unsetenv(name);
+  }
```

Comment:
Let's do this in a separate PR, it's a rather subtle but important change. Is it needed to make some oldtests pass? Can you mention some of those oldtests in the commit message?

Reply:
> Is it needed to make some oldtests pass?

Probably not - Vim does not want to test this behavior.
(there might be an existing one though)

Reply:
Separated into #10743. 

> Is it needed to make some oldtests pass?

I checked it, but there seems to be no old tests that need this other than `Test_WindowsHome()`.

Reply:
Thanks!  I've thought `Test_WindowsHome` also would not need it really, but it uses this method.